traffic_ops/app/db/traffic_vault_migrate/riak.go

```diff
@@ -0,0 +1,749 @@
+	tc.DeliveryServiceSSLKeys
+	Version string
+}
+type riakSSLKeyTable struct {
+	Records []riakSSLKeyRecord
+}
+
+func (tbl *riakSSLKeyTable) gatherKeys(cluster *riak.Cluster, timeout int) error {
+	searchDocs, err := search(cluster, INDEX_SSL, "cdn:*", "", 1000, SCHEMA_SSL_FIELDS[:])
+	if err != nil {
+		return fmt.Errorf("RiakSSLKey gatherKeys: %w", err)
+	}
+
+	tbl.Records = make([]riakSSLKeyRecord, len(searchDocs))
+	for i, doc := range searchDocs {
+		objs, err := getObject(cluster, doc.Bucket, doc.Key, timeout)
+		if err != nil {
+			return err
+		}
+		if len(objs) < 1 {
+			return fmt.Errorf("RiakSSLKey gatherKeys unable to find any objects with key %s and bucket %s, but search results were returned", doc.Key, doc.Bucket)
+		}
+		if len(objs) > 1 {
+			return fmt.Errorf("RiakSSLKey gatherKeys key '%s' more than 1 ssl key record found %d\n", doc.Key, len(objs))
+		}
+		var obj tc.DeliveryServiceSSLKeys
+		if err = json.Unmarshal(objs[0].Value, &obj); err != nil {
+			return fmt.Errorf("RiakSSLKey gatherKeys key '%s' unable to unmarshal object into tc.DeliveryServiceSSLKeys: %w", doc.Key, err)
+		}
+		tbl.Records[i] = riakSSLKeyRecord{
+			DeliveryServiceSSLKeys: obj,
+			Version:                strings.Split(objs[0].Key, "-")[1],
+		}
+	}
+	return nil
+}
+func (tbl *riakSSLKeyTable) toGeneric() []SSLKey {
+	keys := make([]SSLKey, len(tbl.Records))
+
+	for i, record := range tbl.Records {
+		keys[i] = SSLKey{
+			DeliveryServiceSSLKeys: record.DeliveryServiceSSLKeys,
+			Version:                record.Version,
+		}
+	}
+
+	return keys
+}
+func (tbl *riakSSLKeyTable) fromGeneric(keys []SSLKey) {
+	tbl.Records = make([]riakSSLKeyRecord, len(keys))
+
+	for i, record := range keys {
+		tbl.Records[i] = riakSSLKeyRecord{
+			DeliveryServiceSSLKeys: record.DeliveryServiceSSLKeys,
+			Version:                record.Version,
+		}
+	}
+}
+func (tbl *riakSSLKeyTable) insertKeys(cluster *riak.Cluster, timeout int) error {
+	for _, record := range tbl.Records {
+		objBytes, err := json.Marshal(record.DeliveryServiceSSLKeys)
+		if err != nil {
+			return fmt.Errorf("RiakSSLKey insertKeys '%s' failed to marshal keys: %w", record.Key, err)
+		}
+		if err = setObject(cluster, makeRiakObject(objBytes, record.DeliveryService+"-"+record.Version), BUCKET_SSL, timeout); err != nil {
+			return fmt.Errorf("RiakSSLKey insertKeys '%s': %w", record.Key, err)
+		}
+	}
+	return nil
+}
+func (tbl *riakSSLKeyTable) validate() []string {
+	errs := []string{}
+	for _, record := range tbl.Records {
+		if record.DeliveryService == "" {
+			errs = append(errs, fmt.Sprintf("SSL Key '%s': Delivery Service is blank!", record.Key))
+		}
+		if record.CDN == "" {
+			errs = append(errs, fmt.Sprintf("SSL Key '%s': CDN is blank!", record.Key))
+		}
+		if record.Version == "" {
+			errs = append(errs, fmt.Sprintf("SSL Key '%s': Version is blank!", record.Key))
+		}
+	}
+	return errs
+}
+
+type riakDNSSecKeyRecord struct {
+	CDN string
+	Key tc.DNSSECKeysRiak
```

Comment:
nit: `tc.DNSSECKeysRiak` is deprecated, `tc.DNSSECKeysTrafficVault` is the same thing and could be used instead